src/utils/html.rs

```diff
@@ -77,11 +80,33 @@ pub(crate) fn rewrite_lol(
 
     // The input and output are always strings, we just use `&[u8]` so we only have to validate once.
     let mut buffer = Vec::new();
+    // TODO: Make the rewriter persistent?
     let mut writer = HtmlRewriter::try_new(settings, |bytes: &[u8]| {
         buffer.extend_from_slice(bytes);
     })
     .expect("utf8 is a valid encoding");
-    writer.write(html)?;
-    writer.end()?;
+
+    log_memory_errors(writer.write(html), file_path)?;
+    log_memory_errors(writer.end(), file_path)?;
+
     Ok(buffer)
 }
+
+/// Logs OOM errors encountered while rewriting a html file
+fn log_memory_errors<T>(
+    result: Result<T, RewritingError>,
+    file_path: &Path,
+) -> Result<T, RewritingError> {
+    match result {
+        err @ Err(RewritingError::MemoryLimitExceeded(..)) => {
+            crate::web::metrics::HTML_REWRITE_OOMS.inc();
+            log::error!(
```

Comment:
Rather than adding an extra log message here, does chaining the message on via `.context` result in a decent error message from the standard logging?

Reply:
(though I guess that doesn't help with the metrics)

Reply:
It'd probably have roughly the same effect, but this makes sure that an error is always printed, regardless of if the error from the function is caught later on or not